scrapy/http/response/text.py

```diff
@@ -155,19 +160,79 @@ def follow(self, url, callback=None, method='GET', headers=None, body=None,
             cb_kwargs=cb_kwargs,
         )
 
+    def follow_all(self, urls=None, callback=None, method='GET', headers=None, body=None,
+                   cookies=None, meta=None, encoding=None, priority=0,
+                   dont_filter=False, errback=None, cb_kwargs=None,
+                   css=None, xpath=None):
+        # type: (...) -> Generator[Request, None, None]
+        """
+        A generator that produces :class:`~.Request` instances to follow all
+        links in ``urls``. It accepts the same arguments as the :class:`~.Request`'s
+        ``__init__`` method, except that each ``urls`` element does not need to be
+        an absolute URL, it can be any of the following:
+
+        * a relative URL
+        * a :class:`~scrapy.link.Link` object, e.g. the result of
+          :ref:`topics-link-extractors`
+        * a :class:`~scrapy.selector.Selector` object for a ``<link>`` or ``<a>`` element, e.g.
+          ``response.css('a.my_link')[0]``
+        * an attribute :class:`~scrapy.selector.Selector` (not SelectorList), e.g.
+          ``response.css('a::attr(href)')[0]`` or
+          ``response.xpath('//img/@src')[0]``
+
+        In addition, ``css`` and ``xpath`` arguments are accepted to perform the link extraction
+        within the ``follow_all`` method (only one of ``urls``, ``css`` and ``xpath`` is accepted).
+
+        Note that when using the ``css`` or ``xpath`` parameters, this method will not produce
```

Comment:
So, 

```py
yield from response.follow_all(css='a')
```

is not a shortcut for 

```py
sel = response.css('a')
yield from response.follow_all(sel)
```

? The difference is in what happens when some of the links don't have href attribute. With `css='a'` such links are skipped; with `sel` links before the first "a without href" are scheduled, then exception is raised?

Reply:
Nice catch, thanks for noticing. Updated the implementation, docs and tests accordingly.